python/packages/ai/teams/ai/planner/openai_planner.py

```diff
@@ -0,0 +1,172 @@
+            )
+
+            if assistant_prefix:
+                # The model sometimes predicts additional text
+                # for the human side of things so skip that.
+                position = result.lower().index(assistant_prefix.lower())
+                if position >= 0:
+                    result = result[position + len(assistant_prefix) :]
+
+            plan: Plan = parse_response(result)
+
+            # Filter to only a single SAY command
+            if self._options.one_say_per_turn:
+                spoken: bool = False
+                new_commands = []
+                for command in plan.commands:
+                    if command.command_type == CommandType.SAY:
+                        if spoken:
+                            continue
+
+                        new_commands.append(command)
+                        spoken = True
+                    else:
+                        new_commands.append(command)
+                plan.commands = new_commands
+
+        return plan
+
+    def _add_text_completion_service(self) -> None:
+        # TODO: default_model may not be text completion
+        self._sk.add_text_completion_service(
+            "openai_text_completion",
+            OpenAITextCompletion(
+                self._options.default_model, self._options.api_key, self._options.organization
+            ),
+        )
+
+    def _add_chat_completion_service(self) -> None:
+        # TODO: default_model may not be chat completion
+        self._sk.add_chat_service(
+            "openai_chat_completion",
+            OpenAIChatCompletion(
+                self._options.default_model, self._options.api_key, self._options.organization
+            ),
+        )
+
+    async def _complete_prompt(
+        self,
+        prompt_template: PromptTemplate,
+    ) -> str:
+        model: str = self._get_model(prompt_template)
+        is_chat_completion: bool = model.lower().startswith("gpt-")
+        start_time = datetime.now()
+        log_prefix = "CHAT" if is_chat_completion else "PROMPT"
+
+        self._log_request(f"\n{log_prefix} REQUEST: \n'''\n{prompt_template.text}\n'''")
+
+        result: str
+
+        # TODO: implement chat completion
+        result = await self._complete_text(prompt_template)
+
+        duration = datetime.now() - start_time
+        # TODO: investigate how to get prompt/completion tokens
+        self._log_request(f"\n{log_prefix} SUCCEEDED: duration={duration} response={result}")
+
+        return result
+
+    async def _complete_text(self, prompt_template: PromptTemplate):
+        prompt_template_config = generate_sk_prompt_template_config(prompt_template)
+        request_settings = CompleteRequestSettings.from_completion_config(
+            prompt_template_config.completion
+        )
+
+        text_completion_client = self._sk.get_ai_service(TextCompletionClientBase)(self)
+        result = await text_completion_client.complete_async(prompt_template.text, request_settings)
```

Comment:
![image](https://github.com/microsoft/teams-ai/assets/44811138/c9830355-2d76-4787-a855-5557d2e1c8ae)

Reply:
This is not detected by mypy. @aacebo can you investigate?
I will fix this but I think it should be detected in CI